### PR TITLE
BUGFIX: SD-827 SD-843 Handle JS Error If Measure contains same values

### PR DESCRIPTION
**Pull request by dlcd2203**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2830

- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2579
# Related Jira tasks

SD-827: https://jira.intgdc.com/browse/SD-827
SD-843: https://jira.intgdc.com/browse/SD-843

### CODE REVIEW

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6175/ : SUCCESS in 6m 32s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1835/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3193/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1837/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2579/ : SUCCESS in 17m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6201/ : SUCCESS in 12m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1870/ : SUCCESS in 41s

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6202/ : SUCCESS in 7m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1872/ : SUCCESS in 6s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3208/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1873/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2591/ : UNSTABLE in 14m 57s

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6205/ : SUCCESS in 6m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1877/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3210/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1878/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2593/ : SUCCESS in 15m 10s

**Comment by ptquang86** [comment]
LGTM

File: .storybook/test-storybook-canvas.config.js [hunk 1]
```diff
@@ -11,8 +11,10 @@ const configuration = {
             'with location, size and segmentBy': DEFAULT_SCENARIO_CONFIG,
             'with location and color': DEFAULT_SCENARIO_CONFIG,
             'with location, size and color': DEFAULT_SCENARIO_CONFIG,
+            'with location and size, color contains same a value': DEFAULT_SCENARIO_CONFIG,
             'with location, size, color, segmentBy and tooltipText': DEFAULT_SCENARIO_CONFIG,
             'with location, size, color, segmentBy, tooltipText and location filter': DEFAULT_SCENARIO_CONFIG,
+
```
Comment (lvkhanh) [inline .storybook/test-storybook-canvas.config.js]: remove empty space please

**Comment by lvkhanh** [comment]
I have minor comment, otherwise LGTM

File: stories/core_components/GeoPushpinChart.tsx [hunk 2]
```diff
@@ -108,6 +108,21 @@ storiesOf("Core components/GeoPushpinChart", module)
             }),
         ),
     )
+    .add("with location and size, color contains same a value", () => {
```
[the remark this answers is not in the document]
Comment (dlcd2203) [reply stories/core_components/GeoPushpinChart.tsx]: fixed

File: src/components/core/geoChart/tests/geoChartDataLayers.spec.ts [hunk 3]
```diff
@@ -38,6 +39,39 @@ describe("createPushpinDataLayer", () => {
...
+                name: "color",
+            },
+            location: {
+                index: 0,
+                name: "location",
+            },
+        };
+        let excutionResult = getExecutionResult(true, false, false, true, true);
+        const data: Execution.DataValue[][] = [Array(50).fill(1), Array(50).fill(10)];
+        excutionResult = { ...excutionResult, data };
```
[the remark this answers is not in the document]
Comment (dlcd2203) [reply src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: fixed

File: src/components/core/geoChart/tests/geoChartColor.spec.ts [hunk 4]
```diff
@@ -224,15 +224,7 @@ describe("generateLegendColorData", () => {
...
-        expect(colorData).toEqual([
-            {
-                color: "rgb(20,178,226)",
-                range: {
-                    from: 1,
-                    to: 1,
-                },
-            },
-        ]);
+        expect(colorData).toEqual([]);
```
[the remark this answers is not in the document]
Comment (dlcd2203) [reply src/components/core/geoChart/tests/geoChartColor.spec.ts]: fixed

File: .storybook/test-storybook-canvas.config.js [hunk 5]
```diff
@@ -11,8 +11,10 @@ const configuration = {
             'with location, size and segmentBy': DEFAULT_SCENARIO_CONFIG,
             'with location and color': DEFAULT_SCENARIO_CONFIG,
             'with location, size and color': DEFAULT_SCENARIO_CONFIG,
+            'with location and size, color contains same a value': DEFAULT_SCENARIO_CONFIG,
```
[the remark this answers is not in the document]
Comment (dlcd2203) [reply .storybook/test-storybook-canvas.config.js]: fixed

File: mock-schema.js [hunk 6]
```diff
@@ -269,6 +269,11 @@ const geoChartAfmExecutions = [
         executionResponse: require("./stories/test_data/geo_chart/geo_chart_with_location_size_color_response.json"),
         executionResult: require("./stories/test_data/geo_chart/geo_chart_with_location_size_color_result.json")
     },
+    {
+        _description: "GeoPushpin with location, size an color contains same a value",
```
[the remark this answers is not in the document]
Comment (dlcd2203) [reply mock-schema.js]: fixed

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6236/ : FAILURE in 20m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1925/ : SUCCESS in 11s

**Comment by dlcd2203** [comment]
ok to test

**Comment by dlcd2203** [comment]
extended ok to test

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6237/ : FAILURE in 13m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1926/ : SUCCESS in 8s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3234/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1927/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2616/ : UNSTABLE in 21m 42s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by dlcd2203** [comment]
extended test - storybo

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6242/ : SUCCESS in 18m 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1934/ : SUCCESS in 8s

**Comment by yenkins** [comment]
Demo is running at https://client-demo.na.intgdc.com:50025/storybook/

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1935/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2619/ : FAILURE in 11m 03s

**Comment by naibinh** [comment]
We have to run test in AD and KD

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1938/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2620/ : FAILURE in 9m 04s

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6252/ : SUCCESS in 15m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1941/ : SUCCESS in 8s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3243/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1942/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2622/ : SUCCESS in 15m 26s

File: src/components/core/geoChart/tests/geoChartDataLayers.spec.ts [hunk 7]
```diff
@@ -38,6 +39,84 @@ describe("createPushpinDataLayer", () => {
...
+                index: 1,
+                name: "color",
+            },
+            location: {
+                index: 0,
+                name: "location",
+            },
+        };
+        const excutionResult: Execution.IExecutionResult = {
+            data: [Array(3).fill(1), Array(3).fill(10)],
```
Comment (ptquang86) [inline src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: the length of these items should be equal, right?
`[Array(3).fill(10), Array(3).fill(10)],`

File: src/components/core/geoChart/tests/geoChartDataLayers.spec.ts [hunk 8]
```diff
@@ -38,6 +39,84 @@ describe("createPushpinDataLayer", () => {
...
+                                uri: "/gdc/md/projectId/obj/694/elements?id=1810",
+                            },
+                        },
+                    ],
+                ],
+            ],
+            paging: {
+                count: [2, 50],
+                offset: [0, 0],
+                total: [2, 50],
```
Comment (ptquang86) [inline src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: this should equal to [line](https://github.com/gooddata/gooddata-react-components/pull/1399/files#diff-08a085e86141af62510a25c0525d0ffeR58) 
```
count: [2, 10],
offset: [0, 0],
total: [2, 10],
```

**Comment by ptquang86** [comment]
@dlcd2203 I have 2 small comments :)

File: src/components/core/geoChart/tests/geoChartDataLayers.spec.ts (hunk 8 above)
Comment (dlcd2203) [reply src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: fixed

File: src/components/core/geoChart/tests/geoChartDataLayers.spec.ts (hunk 7 above)
Comment (dlcd2203) [reply src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: They are two separate arrays, one for Size and one for Color.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6256/ : SUCCESS in 7m 04s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1947/ : SUCCESS in 5s

File: src/components/core/geoChart/tests/geoChartDataLayers.spec.ts (hunk 7 above)
Comment (dlcd2203) [reply src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: Yes. the length of these items is 3

Comment (ptquang86) [reply src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: my bad :)

**Comment by ptquang86** [comment]
LGTM

**Comment by naibinh** [comment]
@dlcd2203 , we should reuse `/stories/data/geoChart.ts` to generate data for any `_result.js`

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1399.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6258/ : SUCCESS in 7m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1949/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1950/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2624/ : SUCCESS in 41m 44s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3245/ : UNSTABLE in 15m 00s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6260/ : SUCCESS in 5m 02s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1519/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2138/ : SUCCESS in 24m 33s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3246/ : SUCCESS in 17m 45s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6261/ : SUCCESS in 10m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1520/ : SUCCESS in 1m 05s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2139/ : SUCCESS in 21m 23s

**Pull request merged**